### PR TITLE
Bug 1823929 - New ETP UI tests

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Adds some new tests for all ETP options, trackers blocked, adding and removing exceptions.
Also improved the coverage of existing tests.
Worked on improving the stability of tests.
Ran on Firebase multiple times, all green.
Before merging, we should wait for https://github.com/mozilla-mobile/firefox-android/pull/1340 as they might have a conflict.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1823929

### CODE REVIEW

**Review by AndiAJ — APPROVED** [review]
👍🏽

File: fenix/app/src/androidTest/assets/pages/trackingPage.html [hunk 1]
```diff
@@ -2,14 +2,15 @@
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<!-- This asset is using the code behind
+   - https://www.mozilla-anti-tracking.com/test/trackingprotection/test_pages/tracking_protection.html
+   - test page. -->
```
Comment (AaronMT) [inline fenix/app/src/androidTest/assets/pages/trackingPage.html:7]: Mind also adding source repository https://github.com/mozilla/anti-tracking-test-pages

**Review by AaronMT — CHANGES_REQUESTED** [review]
Looks like a minor conflict, otherwise looks OK to me. Lot of refactoring here.

File: fenix/app/src/androidTest/assets/pages/trackingPage.html (hunk 1 above)
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/assets/pages/trackingPage.html]: done

**Pull request merged**